scripts/vendor_prost.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+readonly SCRIPTS_DIR="$(dirname "$0")"
+# shellcheck source=scripts/common
+source "${SCRIPTS_DIR}/common"
+
+# Updates our local copy of https://github.com/danburkert/prost and applies a small patch to add
+# support for the message_type field option.
+
+# Master branch as of 2020-06-05
+PROST_ZIP_URL="https://github.com/danburkert/prost/archive/80256277997975948d257faf3f35c2890bf12787.zip"
+PROST_ZIP_SHA256="02c086f2d1b7aad110745ac701f001618cd8343f882fc9ce034aa007c4c53773"
+PROST_ZIP_PREFIX="prost-80256277997975948d257faf3f35c2890bf12787"
+
+# Make sure we run in the third_party directory.
+cd "${SCRIPTS_DIR}/../third_party/prost" || exit
+
+# Download the prost zip only if it does not yet exist in the filesystem.
+if [ ! -f "$PROST_ZIP_PREFIX.zip" ]; then
+    rm -f "$PROST_ZIP_PREFIX.zip"
+    curl -L "$PROST_ZIP_URL" -o "$PROST_ZIP_PREFIX.zip"
+fi
+
+# Make sure the SHA256 checks out
+echo "$PROST_ZIP_SHA256 $PROST_ZIP_PREFIX.zip" | sha256sum -c -
+
+# Clean up from previous runs
+rm -rf "$PROST_ZIP_PREFIX" vendored/
+
+# Unzip and move into the vendored directory (stripping the prefix).
+unzip "$PROST_ZIP_PREFIX.zip"
+mv "$PROST_ZIP_PREFIX" vendored
```

Comment:
I thought about this for a bit, maybe it's cleaner to put it under `third_party/prost` instead of `third_party/prost/vendored`

Reply:
I thought that too, but didn't want to cause extra merge pain.  If you're happy to include a directory change in your rebase, I'll move it.